brokerapi/brokers/spanner/definition.go

```diff
@@ -63,20 +70,36 @@ func init() {
 			{
 				FieldName: "name",
 				Type:      broker.JsonTypeString,
-				Details:   "The name of the instance.",
-				Default:   "a generated value",
+				Details:   "A unique identifier for the instance, which cannot be changed after the instance is created.",
+				Default:   "pcf-sb-${counter.next()}-${time.nano()}",
+				Constraints: validation.NewConstraintBuilder().
+					MinLength(6).
+					MaxLength(30).
+					Pattern("^[a-z][-a-z0-9]*[a-z0-9]$").
+					Build(),
 			},
 			{
 				FieldName: "display_name",
 				Type:      broker.JsonTypeString,
-				Details:   "A human-readable name for the instance.",
-				Default:   "a generated value",
+				Details:   "The name of this instance configuration as it appears in UIs.",
+				Default:   "${name}",
+				Constraints: validation.NewConstraintBuilder().
+					MinLength(4).
+					MaxLength(30).
+					Build(),
 			},
 			{
 				FieldName: "location",
 				Type:      broker.JsonTypeString,
 				Default:   "regional-us-central1",
-				Details:   `The location of the Spanner instance.`,
+				Details: `A configuration for a Cloud Spanner instance.
+				 Configurations define the geographic placement of nodes and their replication and are slightly different from zones.
+				 There are single region configurations, multi-region configurations, and multi-continent configurations.
+				 See the instance docs for a list of configurations: https://cloud.google.com/spanner/docs/instances.`,
```

Comment:
Consider removing the period at the end of this sentence to prevent users from unwittingly copying/pasting it and getting a 404.